### ISSUE TITLE
Rizin Plugin and snippets hub

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rz-pm#7]
A simple page built on top of the `rz-pm` database allowing to navigate through the packages, maybe show description and dependencies for every package, along with versions and links.

Something of likes
- [Hackage](http://hackage.haskell.org/packages/browse)
- [Opam](https://opam.ocaml.org/packages/)
- etc

![image](https://user-images.githubusercontent.com/203261/78001092-c9a3c380-7367-11ea-9343-c89c160f6825.png)

**Comment by wargio** [comment]
actually i think that can be useful also a zignature repo with known stuff.

**Comment by DRACrY0** [comment]
Is the issue still open?

**Comment by ret2libc** [comment]
It is.

**Comment by yas8say** [comment]
is it still open?

**Comment by wargio** [comment]
yes, it is.

**Comment by yas8say** [comment]
can you assign me this please?

**Comment by yas8say** [comment]
> yes, it is, this could be generated as a static website from the rz-pm database. (we could have this under `packages.rizin.re`)


can you assign me this please?

**Comment by wargio** [comment]
no need, just work on it.

**Comment by yas8say** [comment]
okay.

**Comment by yas8say** [comment]
Quick question: Where will you host the packages on a server or just redirect to the github page ?

**Comment by yas8say** [comment]
> Quick question: Where will you host the packages on a server or just redirect to the github page ?

Quick question: Where will you host the packages on a server or just redirect to the github page ?

**Comment by yas8say** [comment]
> no need, just work on it.

Quick question: Where will you host the packages on a server or just redirect to the github page ? eg: https://github.com/rizinorg/rizin-extras/releases/tag/v0.6.0

**Comment by wargio** [comment]
just redirect the page like we do for the book.rizin.re

**Comment by wargio** [comment]
each one is a plugin that can be installed. you can see each plugin as a package.

**Comment by yas8say** [comment]
got it

**Comment by yas8say** [comment]
@wargio i am using hugo if you dont like the theme i can change it
<img width="761" alt="Screenshot 2024-11-18 at 7 39 37 PM" src="https://github.com/user-attachments/assets/8578f156-2442-4a9a-aa68-4b225da4a260">

**Comment by wargio** [comment]
i'm ok with that as long as is statically generated

**Comment by wargio** [comment]
@XVilka

**Comment by yas8say** [comment]
<img width="1159" alt="Screenshot 2024-11-18 at 7 58 32 PM" src="https://github.com/user-attachments/assets/c2339943-1714-4af6-ac26-c88782ffc770">
These things have to be updated when a package is updated:
Name
Downloads
Rating
Description
Tags
Last U/L
Reference Version
Maintainers 
@wargio How can we do it w/o a db?

**Comment by wargio** [comment]
usually we update them via tag on the package itself. we could just generate a json struct for this each time we push a change on the db repo.

**Comment by XVilka** [comment]
Yes, I recommend sticking with JSON for now, since initially the size of the "database" will not be big. After all, changing the storage backend should be easy once the need arises.

**Comment by yas8say** [comment]
okay @wargio https://rizin.netlify.app/

**Comment by wargio** [comment]
looks good! maybe don't add such big title.

**Comment by yas8say** [comment]
@XVilka @wargio Can you visit the website please and let me know what features/elements should i add or remove?

**Comment by wargio** [comment]
i would prefer if it had a simple search like this https://pkgs.alpinelinux.org/packages because it looks like more a blog rather than a packages search page.

**Comment by yas8say** [comment]
@XVilka @wargio i'll work on it. thanks

**Comment by wargio** [comment]
yes, so much better.

**Comment by yas8say** [comment]
@wargio @XVilka can we automate the json to hugo's md file and push it to the git?

**Comment by wargio** [comment]
You can fork and make a PR to show these changes

**Comment by yas8say** [comment]
@XVilka @wargio guys all the packages are bundled in https://github.com/rizinorg/rizin-extras.git should we separate them for clarity? I can do it.

**Comment by wargio** [comment]
no need to change extras. just use the db repository. over time we will add new plugins.

**Comment by yas8say** [comment]
@wargio So can you review and let me know if there are any changes to be made and are there any other projects you would like me to work on?

**Comment by yas8say** [comment]
@wargio I've added the button and ref link to `packages.rizin.re` 
<img width="1512" alt="Screenshot 2024-11-20 at 11 37 05 AM" src="https://github.com/user-attachments/assets/1140275c-ff58-4fdf-9c97-9fda1995d1de">
What should i do next?

**Comment by XVilka** [comment]
> @wargio So can you review and let me know if there are any changes to be made [rizin.netlify.app](https://rizin.netlify.app/packages/) or are there any other projects you would like me to work on?

There is a lot of empty and unused space in the page top. I recommend reducing/removing it.
Also, align the "Search" field.

**Comment by yas8say** [comment]
@wargio Done

**Comment by yas8say** [comment]
@wargio i've made the PR https://github.com/rizinorg/website/pull/67

**Comment by yas8say** [comment]
@wargio shoud i just point to `rizin.netlify.app` instead of `packages.rizin.re?` . Seems more easier

**Comment by wargio** [comment]
no, we need to build the webiste using the github CI. see how we do this for the main website.

**Comment by yas8say** [comment]
@wargio You need to make a new repo for the website so i can fork and then do PR

**Comment by wargio** [comment]
can't we use the db repo on a new branch?

**Comment by yas8say** [comment]
Okay that will do.

**Comment by yas8say** [comment]
@wargio can you double check : 
`name: Deploy to GitHub Pages

on:
  push:
    branches:
      - master
      - dev
  pull_request:
    branches:
      - master
      - dev

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository
        uses: actions/checkout@v4

      # Install Hugo CLI (latest version)
      - name: Install Hugo CLI
        run: |
          wget -O hugo.deb https://github.com/gohugoio/hugo/releases/latest/download/hugo_extended_$(uname -s)_$(uname -m).deb
          sudo dpkg -i hugo.deb

      # Install Node.js (latest LTS version)
      - name: Install Node.js
        uses: actions/setup-node@v3
        with:
          node-version: "lts"

      # Build the site with Hugo
      - name: Build site with Hugo
        run: |
          hugo --buildDrafts --gc --verbose --baseURL ${{ github.repository }}

      # Run Pagefind
      - name: Run Pagefind
        run: |
          npx pagefind --site public

      # Deploy to GitHub Pages - Stable for master, Dev for dev branch
      - name: Deploy to GitHub Pages
        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: public
          cname: rizin.re

      - name: Deploy Dev to GitHub Pages
        if: github.event_name == 'push' && github.ref == 'refs/heads/dev'
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: public
          cname: dev.rizin.re
`

**Comment by yas8say** [comment]
@wargio made PR.

**Comment by wargio** [comment]
I will do everything this weekend. Sorry for the delay

**Comment by yas8say** [comment]
> I will do everything this weekend. Sorry for the delay

Okay. Take your time no rush :)

**Comment by yas8say** [comment]
Hey @wargio , i wanted to talk about something off-topic can you ping me here at telegram @yas8say

**Comment by wargio** [comment]
You can join our mattermost `im.rizin.re` where we discuss about rizin

**Comment by opcode-poet** [comment]
is the issue still open @wargio

**Comment by opcode-poet** [comment]
hi @wargio , is the issue still open ? and also if possible could i please get to know what are the current hurdles requiring the solution?
